sdk/monitor/azure-monitor-opentelemetry-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/CustomValidationPolicy.java

```diff
@@ -42,15 +41,12 @@ public Mono<HttpResponse> process(
                 .map(CustomValidationPolicy::ungzip);
         asyncBytes.subscribe(
             value -> {
-                ObjectMapper objectMapper = createObjectMapper();
-                try (MappingIterator<TelemetryItem> i =
-                         objectMapper.readerFor(TelemetryItem.class).readValues(value)) {
-                    while (i.hasNext()) {
-                        actualTelemetryItems.add(i.next());
-                    }
-                    countDown.countDown();
+                try (JsonReader jsonReader = JsonProviders.createReader(value)) {
+                    actualTelemetryItems.add(TelemetryItem.fromJson(jsonReader));
```

Comment:
This may have different results than the previous design. `ObjectMapper.readerFor(TelemetryItem.class).readValues(value)` can handle both the single `TelemetryItem` case (`{ }`) and the multiple `TelemetryItem`s case (`[ { }, { }, { }]`).

This will need to be replaced with a helper method like

```java
JsonToken token = jsonReader.currentToken();
if (token == null) {
    token = jsonReader.nextToken();
}

if (token == JsonToken.START_OBJECT) {
    actualTelemetryItems.add(TelemetryItem.fromJson(jsonReader));
} else if (token == JsonToken.START_ARRAY) {
    actualTelemetryItems.addAll(jsonReader.readArray(TelemetryItem::fromJson));
} else {
    // Probably throw an exception as this is an invalid state.
}
```

Reply:
yes! will try it. once i test it working, do you mind i create another github issue for this later?

Reply:
i tried it, it detected token as START_OBJECT only and never hit START_ARRAY.

Reply:
my test creates 1 span at a time, it works. if i create an array of spans, test will fail with my existing implementation. if readArray works, that will solve the problem. I need to divide byte[] into list<byte[]> based on newline delimiters. please let me know if you're good with a new github issue @alzimmermsft

Reply:
`actualTelemetryItems.addAll(jsonReader.readArray(TelemetryItem::fromJson));` didn't work it's due to newlines in the json. jackson json supports newlines since it has a newlinejsonwriter, something like that. 

azure-json doesn't treat json with newlines as a valid json? 

```
{1st item}
{2nd item}
{3rd item}
```

jackson way

`JsonGenerator.setRootValueSeparator(new SerializedString("\n"));`

Reply:
anyhow, i updated the test to support array. test was passing because we only generate 1 element in the list.

Reply:
here is the sample i created to test readArray and feel free to try it.

https://github.com/Azure-Samples/ApplicationInsights-Java-Samples/pull/471
and it threw this exception 

```
Caused by: com.azure.json.implementation.jackson.core.JsonParseException: Illegal character ((CTRL-CHAR, code 31)): only regular white space (\r, \n, \t) is allowed between tokens
 at [Source: (byte[])"\u001F�\u0008\u0000\u0000\u0000\u0000\u0000\u0000�}�AK�0\u0010��˜Ӓ�z�MI\u0017dYO�\u0007EB���`6)�ta)��&iU�`\u0008����x�&p���\u001E)�\u001E\u0018�)uÛۊ�U7�\u0003�m�/iz�\u0000톁��5Q|=��g�uB܋���fM�\u001E��@��8������x�\u0010�wI�0���%��9��i�X��-�b�p����s��ER��\u0006�d!g\u0006Z��\u0016�*��:��,򀕾X��$gp.@r�:}�D^\u001EQ\u000E�i$�ҟN�8�\u000B��\u001C��cB\u001B^'�ޏ�Rb�\u001B�!���\u000C�,�Y�G\u0011�t>\u0001oe���\u0001\u0000\u0000"; line: 1, column: 2]
	at com.azure.json.implementation.jackson.core.JsonParser._constructError(JsonParser.java:2393)
	at com.azure.json.implementation.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:736)
	at com.azure.json.implementation.jackson.core.base.ParserMinimalBase._throwInvalidSpace(ParserMinimalBase.java:714)
	at com.azure.json.implementation.jackson.core.json.UTF8StreamJsonParser._skipWSOrEnd(UTF8StreamJsonParser.java:3059)
	at com.azure.json.implementation.jackson.core.json.UTF8StreamJsonParser.nextToken(UTF8StreamJsonParser.java:758)
	at com.azure.json.implementation.DefaultJsonReader.nextToken(DefaultJsonReader.java:102)
```

Reply:
updated code to support newlines deserialization https://github.com/Azure/azure-sdk-for-java/commit/c9502466e69f264d925c7692d055c4291afc5638